Sources/MCPToolMacros/ToolRegistrationMacro.swift

```diff
@@ -0,0 +1,733 @@
+import SwiftSyntax
```

Comment:
Maintainability: This macro file is large. Consider splitting helpers (property extraction, constraint parsing, schema building, codegen) into separate files/modules to improve readability and testability.